src/cursor/cur_version.c

```diff
@@ -0,0 +1,578 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+#define WT_VERSION_CURSOR_METADATA_FORMAT WT_UNCHECKED_STRING(QQQQQQBBBB)
+/*
+ * __curversion_set_key --
+ *     WT_CURSOR->set_key implementation for version cursors.
+ */
+static void
+__curversion_set_key(WT_CURSOR *cursor, ...)
+{
+    WT_CURSOR *table_cursor;
+    WT_CURSOR_VERSION *version_cursor;
+    uint32_t flags;
+    va_list ap;
+
+    version_cursor = (WT_CURSOR_VERSION *)cursor;
+    table_cursor = version_cursor->table_cursor;
+    va_start(ap, cursor);
+    flags = table_cursor->flags;
+    if (F_ISSET(cursor, WT_CURSTD_RAW))
```

Comment:
Could you please elaborate more on the usage of the `WT_CURSTD_RAW` flag? I don't quite understand why we are using it here.

Reply:
Python api calls to set key, set value, get key, and get value with this flag and do all the packing and unpacking at the python layer. Here is to handle this case.

Reply:
With this flag set, __wt_cursor_set_key will behave differently so we need to temporarily clear this flag.